kernel/ptrace.c

```diff
@@ -44,6 +44,12 @@ static void get_user_regs(struct cpu_state *cpu, struct user_regs_struct_ *user_
 //  user_regs_->xss = cpu->xss;
 }
 
+// Ensure stopped, ptrace locked, etc. before calling this
+static void get_user_regs_and_syscall(struct task *task, struct user_regs_struct_ *user_regs_) {
+    get_user_regs(&task->cpu, user_regs_);
+    user_regs_->orig_eax = task->ptrace.syscall;
```

Comment:
Should this be folded into get_user_regs? I see no other users of get_user_regs.